srml/staking/src/lib.rs

```diff
@@ -519,12 +514,15 @@ decl_module! {
 			promise_month: u32
 		) {
 			let stash = ensure_signed(origin)?;
-			ensure!( promise_month <= 36, "months at most is 36.");
+			ensure!(promise_month <= 36, "months at most is 36.");
 			let controller = Self::bonded(&stash).ok_or("not a stash")?;
-			let ledger = Self::ledger(&controller).ok_or("not a controller")?;
+			let mut ledger = Self::ledger(&controller).ok_or("not a controller")?;
 			match value {
 				 StakingBalance::Ring(r) => {
 					let stash_balance = T::Ring::free_balance(&stash);
+					let expired_locks_ring = T::Ring::update_lock(&stash, None);
+					// TODO: check underflow?
+					ledger.total_ring -= expired_locks_ring;
```

Comment:
Why update expired locks here and remove from total_ring. I remember that unbounding balances is not included in total ring anymore.